alica_engine/src/engine/RuntimePlanFactory.cpp

```diff
@@ -22,11 +22,21 @@ RuntimePlanFactory::RuntimePlanFactory(
 
 void RuntimePlanFactory::init(std::unique_ptr<IPlanCreator>&& pc)
 {
-    _creator = std::move(pc);
+    {
+        std::lock_guard lk(_m);
+        _creator = std::move(pc);
+        _initialized = true;
+    }
+    _cv.notify_all();
 }
 
 std::unique_ptr<BasicPlan> RuntimePlanFactory::create(int64_t id, const Plan* planModel) const
 {
+    std::unique_lock lk(_m);
+    while (!_initialized) {
+        _cv.wait(lk);
```

Comment:
Nit: can use `wait()` with predicate.

Reply:
I believe it could be very deep question. The main difference with predicate is that a predicate is a function, small but function and we also have a call for `wait`, so technically we have 2 calls. I believe it's not a big deal in general for modern CPUs.
For case with while it's just testing one bool w/o calls if it's already in true (like expected in this case)

Reply:
Would also prefer using predicate, if only to make it cleaner.